src/cpp/shared/game/game.h

```diff
@@ -139,7 +140,7 @@ class game {
 	char relatedToHeroForHireStatus[54];
 	mine mines[144];
 	char field_60A6[144];
-	char artifactGeneratedRandomly[103];
+	char artifactGeneratedRandomly[NUM_SUPPORTED_ARTIFACTS];
```

Comment:
You can't change the size of built-in structures. Exception: if you decompile everything that uses that structure (and structures that use it directly), as we did for resourceManager.

What you can do is what we did for hero spells: replace this 103-byte array with a 4-byte pointer to a dynamically-allocated array + 99 bytes of padding, decompile all functions that access this memory, accept that it's hard to not leak memory.

Yes, this sucks.

The good news: I'm only finding 4 functions that access this array: InitRandomArtifacts, GetRandomArtifactId, LoadGame, and SaveGame. And they're all already decompiled.

Reply:
D'oh.  I was hoping I could get away with that because the xml format is somewhat agnostic to the size of the array there.  I'll have to work on this to come up with another approach - I'll look at hero spells for an example.

Reply:
Nope. It's because everything in heroes2.asm is using a hard-coded structure offset. When you mess this up (as we have in the past), you get bugs where saying "foo.x = 1" doesn't actually change foo.x. They're quite nasty to debug.